### PR TITLE
Add Device Support - Blade 15” Base (2021) | RZ09-0369x

### LINKED ISSUE #1442: Support for Razer Blade 15 (2021)
### Device Info

* Name: Razer Blade 15 (2021)
* Product Number: RZ09-0369BEA2
* Model Number: RZ09-0369
### Device Information
```
Bus 001 Device 004: ID 1532:026f Razer USA, Ltd Razer Blade
Couldn't open device, some information will be missing
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x026f 
  bcdDevice            2.00
  iManufacturer           1 
  iProduct                2 
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     159
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0010  1x 16 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
```
```
lrwxrwxrwx 1 root root   9 Feb 16 00:36 usb-Azurewave_USB_Camera_0x0001-event-if00 -> ../event6
lrwxrwxrwx 1 root root   9 Feb 16 00:36 usb-Logitech_USB_Receiver-if02-event-mouse -> ../event7
lrwxrwxrwx 1 root root   9 Feb 16 00:36 usb-Logitech_USB_Receiver-if02-mouse -> ../mouse2
lrwxrwxrwx 1 root root  10 Feb 16 00:36 usb-Razer_Razer_Blade-event-if01 -> ../event13
lrwxrwxrwx 1 root root  10 Feb 16 00:36 usb-Razer_Razer_Blade-event-kbd -> ../event10
lrwxrwxrwx 1 root root  10 Feb 16 00:36 usb-Razer_Razer_Blade-if01-event-kbd -> ../event11
lrwxrwxrwx 1 root root  10 Feb 16 00:36 usb-Razer_Razer_Blade-if02-event-mouse -> ../event17
lrwxrwxrwx 1 root root   9 Feb 16 00:36 usb-Razer_Razer_Blade-if02-mouse -> ../mouse3
```
### Packet Captures

<Please see here as we probably need packet captures>
<https://github.com/openrazer/openrazer/wiki/Reverse-Engineering-USB-Protocol>
Will work on this.

### Synapse Screenshots
<Screenshots of all available options as seen on Razer Synapse>
Will work on this.

### PR DESCRIPTION
**Pull request by jpgls**
Based primarily off of @EtienneMueller 's set up for the 2021 Advanced model.

In my testing this works best with razergenie - polychromatic 0.6.x will recognize the device, but not all options behave as expected.

My main interest in getting this working was simply to be able to set a static color for the keyboard, so I haven't done much/any testing of advanced effects or the like.

Comments/suggestions/edits welcome - thanks!

Fixes #1442

### CODE REVIEW

**Comment by terricain** [comment]
Cool so what works and what doesnt that you think should be working?

**Comment by jpgls** [comment]
In Polychromatic - the device is recognized, and the system tray menu works, but if you open the polychromatic controller window, there are no controls available through that interface.

**Comment by jpgls** [comment]
Screenshot from 2021-06-23 20-17-24

Re-installed polychromatic and grabbed a screenshot

**Comment by z3ntu** [comment]
You can also try running RazerGenie to see if the behavior is any different. The patch diff looks sane.

**Comment by oddbunchmedia** [comment]
@jpgls - I'd love to see this working - is the 2021 Base basically the same as the 2020 Base model? If so, maybe it would be better to align changes with that model rather than the Advanced?

File: daemon/openrazer_daemon/hardware/keyboards.py [hunk 1]
```diff
@@ -1306,3 +1306,22 @@ class RazerBlade15AdvancedEarly2021(_RippleKeyboard):
...
+class RazerBlade15BaseEarly2021(_RippleKeyboard):
+    """
+    Class for the Razer Blade 15 Base (Early 2021)
+    """
+    EVENT_FILE_REGEX = re.compile(r'.*Razer_Blade(-if01)?-event-kbd')
+
+    USB_VID = 0x1532
+    USB_PID = 0x026F
+    HAS_MATRIX = True
+    MATRIX_DIMS = [6, 16]
```
Comment (oddbunchmedia) [inline daemon/openrazer_daemon/hardware/keyboards.py:1320]: 2020 Base model is MATRIX_DIMS = [1, 16]

File: daemon/openrazer_daemon/hardware/keyboards.py [hunk 2]
```diff
@@ -1306,3 +1306,22 @@ class RazerBlade15AdvancedEarly2021(_RippleKeyboard):
...
+    METHODS = ['get_device_type_keyboard', 'get_logo_active', 'set_logo_active', 'set_wave_effect', 'set_static_effect', 'set_spectrum_effect',
```
Comment (oddbunchmedia) [inline daemon/openrazer_daemon/hardware/keyboards.py:1321]: 2020 Base model has no logo controls - maybe other differences I only had a quick look

File: driver/razerkbd_driver.c [hunk 3]
```diff
@@ -1994,6 +2000,7 @@ static ssize_t razer_attr_write_matrix_custom_frame(struct device *dev, struct d
         case USB_DEVICE_ID_RAZER_BLADE_PRO_LATE_2019:
         case USB_DEVICE_ID_RAZER_BLADE_15_ADV_2020:
         case USB_DEVICE_ID_RAZER_BLADE_15_ADV_EARLY_2021:
+        case USB_DEVICE_ID_RAZER_BLADE_15_BASE_EARLY_2021:
```
Comment (oddbunchmedia) [inline driver/razerkbd_driver.c:2003]: 2020 Base model isn't in this list, so maybe 2021 shouldn't either?

**Comment by CasperBruning** [comment]
> Screenshot from 2021-06-23 20-17-24
> 
> Re-installed polychromatic and grabbed a screenshot

my razer blade 15 base early 2021 keyboard isnt even recognized. and i have the latest stuff installed

**Comment by oddbunchmedia** [comment]
My Base 2021 is feeling unloved... :(

**Comment by z3ntu** [comment]
Closing in favor of #1653, thanks for your contributions!

**Pull request closed without merging**